src/conn/conn_tiered.c

```diff
@@ -203,14 +204,19 @@ __tier_do_operation(WT_SESSION_IMPL *session, WT_TIERED *tiered, uint32_t id, co
     const char *cfg[2], *local_name, *obj_name, *sp_obj_name;
 
     WT_ASSERT(session, (op == WT_TIERED_WORK_FLUSH || op == WT_TIERED_WORK_FLUSH_FINISH));
+    dhandle = (WT_DATA_HANDLE *)tiered;
     tmp = NULL;
     /*
-     * It is possible that the tiered object was closed before the work unit was processed. The work
-     * unit holds a reference on the dhandle but if the bucket storage is gone there is nothing to
-     * do.
+     * The work unit holds a reference on the dhandle so that the structure is valid to look at, but
+     * the dhandle could have been dropped or under exclusive use. If it isn't open or it is dropped
+     * there is nothing to do. If the bucket storage is gone there is nothing to do. (see if this is
```

Comment:
I don't understand the parenthetical comment here. Can you clarify it?

Reply:
That should be removed.